lib/components/Button/Button.scss

```
@@ -111,3 +82,152 @@ $size-map: (
         }
     }
 }
+
+.deriv-button__variant--contained {
+    &.deriv-button__color--primary {
+        background-color: $primary;
+        span {
+            color: $white;
+        }
+        &:hover:not(:disabled) {
+            background-color: $primary-hover;
+        }
+    }
+}
+
+.deriv-button__variant--contained {
+    &.deriv-button__color--primary-light {
+        background-color: $primary-light;
+        span {
+            color: $primary;
+        }
+        &:hover:not(:disabled) {
+            background-color: $primary-light-hover;
+        }
+    }
+}
+
+.deriv-button__variant--contained {
+    &.deriv-button__color--white {
+        background-color: $white;
+        span {
+            color: $black;
+        }
+        &:hover:not(:disabled) {
+            background-color: $white-hover;
+        }
+    }
+}
+
+.deriv-button__variant--contained {
+    &.deriv-button__color--black {
+        background-color: $black;
+        span {
+            color: $white;
+        }
+        &:hover:not(:disabled) {
+            background-color: $black-hover;
+        }
+    }
+}
```

can we wrap all of the individual variants inside each of the parent variants? something like this?

```suggestion
.deriv-button__variant--contained {
    &.deriv-button__color--primary {
        background-color: $primary;
        span {
            color: $white;
        }
        &:hover:not(:disabled) {
            background-color: $primary-hover;
        }
    }
    
    &.deriv-button__color--primary-light {
        background-color: $primary-light;
        span {
            color: $primary;
        }
        &:hover:not(:disabled) {
            background-color: $primary-light-hover;
        }
    }
    
     &.deriv-button__color--white {
        background-color: $white;
        span {
            color: $black;
        }
        &:hover:not(:disabled) {
            background-color: $white-hover;
        }
    }
    
    &.deriv-button__color--black {
        background-color: $black;
        span {
            color: $white;
        }
        &:hover:not(:disabled) {
            background-color: $black-hover;
        }
    }
}
```

---

same with the others below 👀

---

@aizad-deriv can we use this ?
`@mixin deriv-button-contained($background-color, $text-color, $hover-background-color) {
  background-color: $background-color;
  span {
    color: $text-color;
  }
  &:hover:not(:disabled) {
    background-color: $hover-background-color;
  }
}

.deriv-button__variant--contained {
  &.deriv-button__color--primary {
    @include deriv-button-contained($primary, $white, $primary-hover);
  }

  &.deriv-button__color--primary-light {
    @include deriv-button-contained($primary-light, $primary, $primary-light-hover);
  }

  &.deriv-button__color--white {
    @include deriv-button-contained($white, $black, $white-hover);
  }
}`